src/spikeinterface/core/basesorting.py

```diff
@@ -449,6 +451,33 @@ def get_all_spike_trains(self, outputs="unit_id"):
             spikes.append((spike_times, spike_labels))
         return spikes
 
+    def precompute_spike_trains(self, from_spike_vector=None):
```

Comment:
I do not like too much this method name.
Maybe `cache_all_spike_trains()` would be more explicit

Reply:
To me, I would expect a function `cache_all_spike_trains` to not perform any computation, but just move the result of computation to the cache.
But I agree that the name could maybe be improved